### ISSUE TITLE
ivy: accessing undefined ViewChild properly with elvis operator in template breaks

### ISSUE THREAD

**Issue by mattlewis92** [angular/angular#34087]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/core

### Is this a regression?
Yes, works fine in v8

### Description
If I add a `ViewChild` decorator to an undefined component in a template and try and access it using the elvis operator in a template then the expression breaks and nothing is outputted in the template.

## 🔬 Minimal Reproduction
https://github.com/mattlewis92/ivy-view-child-accesssor-bug/blob/master/src/app/app.component.html#L1

Where foo is:
https://github.com/mattlewis92/ivy-view-child-accesssor-bug/blob/master/src/app/app.component.ts#L11

## 🔥 Exception or Error
There is no error, it fails silently to render:
![Screenshot 2019-11-27 at 15 50 48](https://user-images.githubusercontent.com/6425649/69738255-b5d42080-112d-11ea-8dfb-1a926fdd8e16.png)

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 9.0.0-rc.3
Node: 12.10.0
OS: darwin x64
Angular: 9.0.0-rc.3
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-rc.3
@angular-devkit/build-angular     0.900.0-rc.3
@angular-devkit/build-optimizer   0.900.0-rc.3
@angular-devkit/build-webpack     0.900.0-rc.3
@angular-devkit/core              9.0.0-rc.3
@angular-devkit/schematics        9.0.0-rc.3
@ngtools/webpack                  9.0.0-rc.3
@schematics/angular               9.0.0-rc.3
@schematics/update                0.900.0-rc.3
rxjs                              6.5.3
typescript                        3.6.4
webpack                           4.41.2
</code></pre>

**Anything else relevant?**
Nope

**Comment by AndrewKushnir** [comment]
Hi @mattlewis92, thanks for reporting this issue.

I performed an investigation and found out that the code generated by Ivy Compiler in a `template` function looks correct (compiled locally):
```
  textInterpolate1('This doesn\'t work: ',(((ctx.foo == null)? null: ctx.foo.bar)? 'bar': 'bam'),'');
```
and it looks like parentheses are removed after further processing (presumably in CLI):
```
  _angular_core__WEBPACK_IMPORTED_MODULE_0__["ɵɵtextInterpolate1"]("This doesn't work: ", ctx.foo == null ? null : ctx.foo.bar ? "bar" : "bam", " ");
```
Which changes the logic and makes the code return `null` (and render nothing) instead of rendering 'bam' string.

@filipesilva do you know if this is an optimization step in CLI that might affect this?

**Comment by AndrewKushnir** [comment]
FYI @alxhub performed further investigation and fixed the problem in PR #34221 👍
The problem was that I looked at the JIT output and the problem was present in AOT output (which is now aligned with JIT).
Thank you.

**Comment by mattlewis92** [comment]
Amazing, thank you for looking into it so quickly! ❤️
